lib/clients/rtm/client.js

```diff
@@ -441,8 +441,10 @@ RTMClient.prototype._handleWsMessageInternal = function _handleWsMessageInternal
   } else if (messageType === RTM_API_EVENTS.HELLO) {
     this._handleHello();
   } else if (messageType === RTM_API_EVENTS.TEAM_MIGRATION_STARTED) {
-    this.reconnect();
     this.emit(messageType, message);
+    if (this.autoReconnect) {
+      this.reconnect();
+    }
```

Comment:
this will change the order in which the observable events arrive, but i think its more consistent to fire the message before firing something that (might) happen as a result of that message.